commons-kafka/src/main/scala/com/avsystem/commons/kafka/SerdeRegistry.scala

```diff
@@ -0,0 +1,33 @@
+package com.avsystem.commons
+package kafka
+
+import com.avsystem.commons.kafka.exceptions.UnsupportedVersionEvent
+import com.avsystem.commons.serialization.GenCodec
+
+import scala.collection.mutable
+
+
+class SerdeRegistry[T] extends AbstractDeserializer[T] {
+  private val registry = mutable.Map[Byte, KafkaSerde[_ <: T]]()
+
+  def add[X <: T](version: Byte, serde: KafkaSerde[X]): this.type = {
+    registry += (version -> serde)
+    this
+  }
+
+  def add[X <: T](version: Byte)(implicit codec: GenCodec[T]): this.type = {
```

Comment:
Ugh, that generic `X` seems redundant. I overlooked it.

Reply:
yeap, sure :)